docs/tutorials/typescript/work-queue-slack-app/index.mdx

```diff
@@ -0,0 +1,991 @@
+This could especially be true if that object needs to be accessed by a distributed set of services.
+
+### Craft the Entity pattern
+
+The Temporal TypeScript SDK comes with a handy `condition` API.
+You can use this to await on a specific condition before continuing.
+
+In this case the Workflow awaits on a [Continue-As-New](https://docs.temporal.io/workflows#continue-as-new) suggestion from the Temporal Service.
+This is a suggestion to essentially create a new instance of the Workflow with a fresh Event History because the current one is too large.
+When we continue the Workflow as a new one, we pass the existing Work Queue data to the new Workflow instance.
+
+<!--SNIPSTART typescript-slack-app-temporal-application-workqueue-workflow {"selectedLines": ["1-5", "9-10", "21-22", "51-66"]}-->
+
+[docs/tutorials/typescript/work-queue-slack-app/code/temporal-application/src/workflows/workqueue.ts](https://github.com/temporalio/temporal-learning/blob/edu-2636-workqueue-slackbot-tut/docs/tutorials/typescript/work-queue-slack-app/code/temporal-application/src/workflows/workqueue.ts)
+
+```ts
+import {
+  isCancellation,
+  continueAsNew,
+  workflowInfo,
+  condition,
+  // ...
+} from "@temporalio/workflow";
+import {WorkqueueData} from "../../../common-types/types";
+// ...
+export async function workqueue(existingData?: WorkqueueData[]): Promise<void> {
+  const wqdata: WorkqueueData[] = existingData ?? [];
+  // ...
+  try {
+    // Await until suggestion to Continue-As-New due to History size
+    // If a Cancellation request exists, the condition call will throw the Cancellation error
+    await condition(() => workflowInfo().continueAsNewSuggested);
+  } catch (e) {
+    // Catch a Cancellation error
+    if (isCancellation(e)) {
+      // Set the Workflow status to Cancelled by throwing the CancelledFailure error
+      throw e;
+    } else {
+      // Handle other types of errors
+      throw e;
+    }
+  }
+  await continueAsNew<typeof workqueue>(wqdata);
+}
+```
+
+<!--SNIPEND-->
+
+What is also handy about the `condition` API is that if a Cancellation request is made to the Workflow, the condition call will throw a Cancellation error.
+
+In the context of Temporal, the code above will create a Workflow Execution that stays running until it is explicitly told to stop.
+
+### Send data in and out of the Workflow
+
+Next, you will want to define and handle Signals and Queries.
+
+**[Signals](https://docs.temporal.io/encyclopedia/application-message-passing#signals)** are a way to send data into a Workflow.
+**[Queries](https://docs.temporal.io/encyclopedia/application-message-passing#queries)** are a way to read the state of a Workflow.
+
+It is a good practice to define the Signals and Queries outside of the Workflow and handle them right after declaring local state variables within the Workflow.
```

Comment:
This section is good (and I'd keep it as an h3), my only comment is I don't have a clear idea of *what* signals and queries are actually being sent to this application at this point, only that it has to talk to Slack.